applications/GeoMechanicsApplication/custom_elements/stress_state_policy.h

```diff
@@ -0,0 +1,36 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//                   Marjan Fathian
+//
+#pragma once
+
+#include "geo_mechanics_application_constants.h"
+#include "geometries/geometry.h"
+#include "includes/node.h"
+#include "includes/ublas_interface.h"
+
+namespace Kratos
+{
+
+class StressStatePolicy
+{
+public:
+    virtual Matrix CalculateBMatrix(const Matrix& GradNpT, const Vector& Np, const Geometry<Node>& rGeometry) const = 0;
```

Comment:
To comply with the Kratos Style Guide, we need to add a few `r` prefixes:
```suggestion
    virtual Matrix CalculateBMatrix(const Matrix& rGradNpT, const Vector& rNp, const Geometry<Node>& rGeometry) const = 0;
```

Also, I'm not sure whether we should add a `p` to the shape function names (e.g. `rGradNpT`, rather than `rGradNT`), since it suggests that these are the shape functions for the water pressures, whereas they would be equally valid if we had used the shape functions of the displacements. Just `N` would be more neutral in that respect.

Reply:
I don't know enough about this, so I tend to just agree based on your reasoning. What do you think @WPK4FEM?